File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.py
class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        jobs = []
        h = []
        for d, p in zip(difficulty, profit):
            heapq.heappush(jobs, (d, p))
            
        worker.sort()
        answer = 0
        for ability in worker:
            while jobs and jobs[0][0] <= ability:
                d, p = heapq.heappop(jobs)
                heapq.heappush(h, -p)
            if h:
                answer += -h[0]
        return answer

File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.rs
use std::collections::BinaryHeap;

impl Solution {
    pub fn max_profit_assignment(difficulty: Vec<i32>, profit: Vec<i32>, worker: Vec<i32>) -> i32 {
        let mut jobs: BinaryHeap<(i32, i32)> = BinaryHeap::new();
        let mut h: BinaryHeap<i32> = BinaryHeap::new();
        let mut answer = 0;
        for (d, p) in difficulty.iter().zip(&profit) {
            jobs.push((-d, *p));
        }
        
        let mut worker = worker.clone();
        worker.sort();
        for ability in worker {
            while !jobs.is_empty() && -jobs.peek().unwrap().0 <= ability {
                let (d, p) = jobs.pop().unwrap();
                h.push(p);
            }
            
            if !h.is_empty() {
                answer += h.peek().unwrap();
            }
        }
        answer
    }
}